### ISSUE TITLE
'default' is not defined in typeof import

### ISSUE THREAD

**Issue by inferusvv** [typescript-eslint/typescript-eslint#3006]
**Repro**

```JavaScript
module.exports = {
  root: true,
  extends: '@react-native-community',
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
};

```

```TS
export type RootAction = ActionType<typeof import('./src/store/actions').default>;
export type AppState = ReturnType<typeof import('./src/store/reducers').default>;
```

**Expected Result**

No errors here like it was before

**Actual Result**

```
/.../types.d.ts
  4:74  error  'default' is not defined  no-undef
  5:73  error  'default' is not defined  no-undef
```

**Additional Info**

In this files (actions and reducers) I have default export

```
const appReducer = combineReducers({
// ...
});

export default appReducer;
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.14.2` |
| `TypeScript`                | `4.1.3` |
| `ESLint`                    | `7.19.0` |
| `node`                      | `12.18.4` |
